test/e2e/framework/service/jig.go

```diff
@@ -82,6 +96,66 @@ func NewTestJig(client clientset.Interface, namespace, name string) *TestJig {
 	return j
 }
 
+// allocate tries to allocate the passed node port and
+// return true if succeeds or false if can not be allocated.
+// Should not be used directly as it does not have lock protection and hence avoids deadlock.
+// Should only be called from a function that has acquired lock on nodeports
+func (s *staticPortRange) allocate(port int) bool {
```

Comment:
I see, so allocate() is only called from allocateNext

I prefer to not have two different ways of passing the nodeport, in this case this receives the relative port and not the absolute, so maybe you just remove this helper function and put all the logic inside allocateNext?

Reply:
yeah, we can remove this entire methos with just two new lines

https://github.com/kubernetes/kubernetes/pull/127153/files#r1756262913